hbase-server/src/main/java/org/apache/hadoop/hbase/rsgroup/RSGroupInfoManagerImpl.java

```diff
@@ -591,7 +591,7 @@ private synchronized void refresh(boolean forceOnline) throws IOException {
 
     // This is added to the last of the list so it overwrites the 'default' rsgroup loaded
     // from region group table or zk
-    groupList.add(new RSGroupInfo(RSGroupInfo.DEFAULT_GROUP, getDefaultServers()));
+    groupList.add(new RSGroupInfo(RSGroupInfo.DEFAULT_GROUP, getDefaultServers(groupList)));
```

Comment:
refresh fetching info from rsgroup table / ZK makes sense.

Reply:
So the issue is by this time, the in memory state 'holder' not created in the HM restart case. This is a very bad bug.. This should go into all active branches no?

Reply:
branch-1 fix was pending, raised PR https://github.com/apache/hbase/pull/2102